docs/zk_security.md

```diff
@@ -0,0 +1,56 @@
+# ZooKeeper security
+
+To prevent accidental or malicious tampering with the data that Helios stores in ZooKeeper Helios
+supports using ZooKeeper's ACL functionality to lock it down (to an extent).  This functionality is
+disabled by default.
+
+When enabled, credentials are required to access data in ZooKeeper (we use ZooKeeper's `digest`
+authentication scheme). There are two sets of credentials: one set of the masters, and one set for 
+the agents. With ACLs enabled unauthenticated users have no access to the data in ZooKeeper (not
+even read access). Agents have read access to all data but limited permissions to mutate data. They
+only have mutate permissions where needed. This limits the impact of the agent credentials being
+compromised (e.g. if an agent is compromised). While the agent credentials being compromised is
+obviously not good the lower privileges they provide prevent an attacker from deploying malicious
+jobs to other agents (in most cases).
+
+Masters are granted all permissions except ADMIN on all nodes.
+
+**Note that credentials are sent to ZooKeeper in plain-text** (we currently don't support using
```

Comment:
Cool, so I think it would be good to link to that page here and change the doc to say that when not using SSL the credentials are sent in plaintext, and if you want to use SSL read this link